docs/docs/setup/auth/user/callback.md

```diff
@@ -0,0 +1,232 @@
+---
+sidebar_position: 2
+---
+
+# Auth callback
+
+> ```toml
+> auth.source = "callback:http://localhost:9090/"
+> ```
+
+## How it works
+
+In this mode, your custom auth logic sits behind Tobira and is called for every incoming request that needs authentication:
+
+```
+┌──────┐                    ┌─────────┐                    ┌────────┐  request headers   ┌────────┐
+│      │ -----------------> │ reverse │ -----------------> │        │ -----------------> │  Your  │
+│ User │                    │  proxy  │                    │ Tobira │                    │  Auth  │
+│      │ <----------------- │         │ <----------------- │        │ <----------------- │  Logic │
+└──────┘                    └─────────┘                    └────────┘   auth/user info   └────────┘
+```
+
+Your callback needs to be an HTTP endpoint that you specify behind the colon of `callback:`.
+The URL can have a path, but no query or fragment part.
+You also have to specify `auth.callback.relevant_headers` and/or `relevant_cookies`: a list of headers/cookies that your auth logic reads.
+
+On receiving a request that needs authentication, Tobira sends a request to your callback.
+That request has no body, but all `relevant_headers` and `relevant_cookies` copied from the incoming request.
+If you include `"cookie"` in the `relevant_headers` field, all cookies are always forwarded (note: this would usually make the caching fairly useless, so you probably want to disable it then).
```

Comment:
Maybe link to the caching section here so I don't have to wonder whether I overread something.